scipy/interpolate/tests/test_bsplines.py

```diff
@@ -3564,6 +3564,32 @@ def test_s0_vs_not(self):
         assert spl_0.t.shape[0] == n + k + 1
         assert spl_1.t.shape[0] == 2 * (k + 1)
 
+    def test_periodic_with_periodic_data(self):
+        N = 10
+        a, b, dx = 0, 2*np.pi, 0.2*np.pi
+        x = np.linspace(a, b, N + 1)    # nodes
+
+        y = np.cos(x)
+        tck = make_splrep(x, y, s=1e-8, periodic=True)
+        xp_assert_close(splev(x, tck), y, atol=1e-5, rtol=1e-4)
+
+        y = np.sin(x) + np.cos(x)
+        tck = make_splrep(x, y, s=1e-12, periodic=True)
+        xp_assert_close(splev(x, tck), y, atol=1e-5, rtol=1e-6)
+
+        y = 5*np.sin(x) + np.cos(x)*3
+        tck = make_splrep(x, y, s=1e-8, periodic=True)
+        xp_assert_close(splev(x, tck), y, atol=1e-5, rtol=1e-4)
```

Comment:
Overall this test looks nice but leaves a question of why s is always this small. I'd add some more tests to probe the cases of 1) no internal knots, just the initial estimate, 2) some internal knots, not all data, 3) smaller s, more knots, finally, all data --- then the result should agree with CubicSpline to a tight tolerance, I'd expect.

Also: you only test k=3. Does splrep(x, y, periodic=True) handle k=1 and/or k=2? If yes, these should be tested (careful with k=2: for interpolation generate_knots switches from data points to Greville abscissae); if not, these should raise fast. [if it's hard to implement, it's fine to bail out and leave a comment that it's not implemented and what is hard].